### PR TITLE
feat(clickup-cdk): Add new codecov bypass workflow [CLK-464388]

### PR DESCRIPTION
**Pull request by m-szaf**
## Summary

When `codecov` service is degraded, all repos are blocked due to their required checks on `codecov/project` or `codecov/patch`. 

This functionality already exists in other repos. Adding this here so all downstream cdk repos can benefit and unblock themselves when needed!

### CODE REVIEW

**Review by slinclickup — APPROVED** [review]

File: src/codecov-bypass-workflow.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { YamlFile } from 'projen';
+import { clickupCdk } from './clickup-cdk';
+
+const DEFAULT_CODECOV_GITHUB_APP_ID = 254;
```
Comment (ahammond) [inline src/codecov-bypass-workflow.ts:4]: Please document the magic number. Maybe a link into GitHub?

File: src/clickup-cdk.ts [hunk 2]
```diff
@@ -232,6 +241,9 @@ export module clickupCdk {
           cdkDiffWorkflow.addOidcRoleStack(this);
         }
       }
+      if (options.codecovBypassOptionsConfig) {
+        codecovBypassWorkflow.addCodecovBypassWorkflowYml(this, options.codecovBypassOptionsConfig);
+      }
```
Comment (ahammond) [inline src/clickup-cdk.ts:246]: +1 go with "explicit enable" to prove it and then we can switch to "enabled by default with explicit disable" and update our README.md, as well as our compliance docs around this.

File: src/codecov-bypass-workflow.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            },
+          ],
+        },
+      },
+    };
+
+    return defaultWorkflow;
+  }
+
+  export interface CodecovBypassOptionsConfig {
```
Comment (ahammond) [inline src/codecov-bypass-workflow.ts:49]: Please add an `enabled` or similar option.

File: src/codecov-bypass-workflow.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    /**
+     * Workflow Name
+     */
```
Comment (ahammond) [inline src/codecov-bypass-workflow.ts:52]: If it's optional, please include a `@default foo` in the docstring.

**Review by ahammond — COMMENTED** [review]
Needs a `docs/codecov-bypass-workflow/README.md`, please.

File: src/codecov-bypass-workflow.ts (hunk 1 above)
Comment (mattlewis92) [reply src/codecov-bypass-workflow.ts:4]: For context, it's the ID on the github codecov app: https://github.com/apps/codecov

File: src/codecov-bypass-workflow.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+export module codecovBypassWorkflow {
+  function CreateCodecovBypassWorkflow(options?: CodecovBypassOptionsConfig) {
+    const defaultWorkflow = {
+      name: options?.workflowName || 'Code coverage increased',
+      on: {
+        check_suite: {
+          types: ['completed'],
+        },
+        pull_request: {
+          types: ['labelled', 'unlabeled'],
```
Comment (mattlewis92) [inline src/codecov-bypass-workflow.ts:15]: `labelled` -> `labeled`

File: src/codecov-bypass-workflow.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        },
+      },
+      permissions: {},
+      jobs: {
+        'code-coverage-increased': {
+          'runs-on': 'ubuntu-latest',
+          if: `github.event_name == \'check_suite\' || github.event.check_suite.app.id == ${options?.CodecovGithubAppId || DEFAULT_CODECOV_GITHUB_APP_ID}`,
```
Comment (mattlewis92) [inline src/codecov-bypass-workflow.ts:22]: `github.event_name != \'check_suite\'`

File: src/codecov-bypass-workflow.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+              uses: 'time-loop/github-actions/dist/wrap-check-suite@wrap-check-suite+0.3.2',
+              env: {
+                FORCE_COLOR: 3,
+              },
+              with: {
+                'github-app-id': options?.githubAppId || '${{ secrets.CLICKUP_GITHUB_BOT_APP_ID }}',
+                'github-private-key': options?.githubAppPrivateKey || '${{ secrets.CLICKUP_GITHUB_BOT_PRIVATE_KEY }}',
+                'skip-label': options?.skipLabel || 'code coverage not required',
+                'check-name': options?.checkName || 'Code coverage increased',
+                'check-suite-app-id': options?.CodecovGithubAppId || DEFAULT_CODECOV_GITHUB_APP_ID,
```
Comment (mattlewis92) [inline src/codecov-bypass-workflow.ts:35]: Not sure that `CodecovGithubAppId` really makes sense here, as we're specifically wrapping the codecov github app

File: src/codecov-bypass-workflow.ts (hunk 1 above)
Comment (m-szaf) [reply src/codecov-bypass-workflow.ts:4]: Ah sweet - that was the missing piece I also didn't know where the number came from

File: src/codecov-bypass-workflow.ts (hunk 3 above)
Comment (m-szaf) [reply src/codecov-bypass-workflow.ts:49]: When enabled: false - should we delete/not render the workflow or what did you have in mind?

**Pull request merged**